src/base/utils/compare.h

```diff
@@ -29,6 +29,7 @@
 #pragma once
 
 #include <Qt>
+#include <QtCore/private/qtcore-config_p.h>  // for QT_FEATURE_xxx
 #include <QtSystemDetection>
 
 #if !defined(Q_OS_WIN) && (!defined(Q_OS_UNIX) || defined(Q_OS_MACOS) || defined(QT_FEATURE_icu))
```

Comment:
Since we're here, could you remind why we don't use QCollator under Windows? I didn't find anything relevant about this in the documentation (or I missed something).

Reply:
>Since we're here, could you remind why we don't use QCollator under Windows?

Found the PR from git blame: https://github.com/qbittorrent/qBittorrent/pull/17204

The conditionals seems to have changed in Qt: https://github.com/qt/qtbase/blob/6.6/src/corelib/CMakeLists.txt#L791-L812
And therefore I would like to change this line to:
```suggestion
#ifdef QT_FEATURE_icu
```

Reply:
> Still, I would like to change this line to: `#ifdef QT_FEATURE_icu`

https://doc.qt.io/qt-6/qcollator.html#posix-fallback-implementation
>However, if ICU was not available at compile time or explicitly disabled, Qt will use a fallback backend that uses the POSIX API only. 

If I understand correctly, we should only detect `QT_FEATURE_icu` when deciding whether to use QCollator or not. The OS is not relevant here.

Reply:
> why we don't use QCollator under Windows? I didn't find anything relevant about this in the documentation (or I missed something).

I mean that I didn't find in documentation anything about the fact that QCollator requires ICU under Windows. Maybe it uses some OS specific API (like it does under macOS)?

Reply:
>Maybe it uses some OS specific API (like it does under macOS)?

It is true. I've updated the commit. See the commit message.